spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -303,13 +304,13 @@ public RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
 		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
+
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
-						: MimeTypeUtils.TEXT_PLAIN;
+				return originalContentType.toLowerCase().contains(MimeTypeUtils.APPLICATION_JSON.getSubtype()) ? MimeType.valueOf(originalContentType) : MimeTypeUtils.TEXT_PLAIN;
```

Comment:
I tend to agree that if the contentType is **any** json type, it should be retained (and that will avoid carrying an OCT header at all, since that's only added if it's different to the CT).

One problem I see with not setting the CT to `text/plain` for a `String` payload is the consumer might have a `@StreamListener` with parameter `Message<?>` and be expecting `Message<String>` but he'll get `Message<byte[]>`. But, as @mbogoevici has pointed out, we could document that in 2.0, if they want `Message<String>`, then the parameter should be `Message<String>`, not `Message<>` (or raw `Message`).